src/test/java/apoc/load/relative/LoadXmlRelativePathTest.java

```diff
@@ -0,0 +1,248 @@
+
+                    List<Object>  childrenList = (List<Object>) children;
+                    Map<String, String> childrenMap = ((Map<String, String>) childrenList.get(0));
+
+                    assertEquals("bk101", result.get("id"));
+                    assertEquals("author", childrenMap.get("_type"));
+                    assertEquals("Gambardella, Matthew", childrenMap.get("_text"));
+                    assertEquals("author", ((Map) childrenList.get(1)).get("_type"));
+                    assertEquals("Arciniegas, Fabio", ((Map) childrenList.get(1)).get("_text"));
+
+                    result = (Map<String, Object>) r.next().get("result");
+                    children = result.get("_children");
+                    childrenList = (List<Object>) children;
+                    childrenMap = ((Map<String, String>) childrenList.get(0));
+
+                    assertEquals("bk110", result.get("id"));
+                    assertEquals("author", childrenMap.get("_type"));
+                    assertEquals("O'Brien, Tim", childrenMap.get("_text"));
+                    assertEquals("title", ((Map) childrenList.get(1)).get("_type"));
+                    assertEquals("Microsoft .NET: The Programming Bible", ((Map) childrenList.get(1)).get("_text"));
+
+                    result = (Map<String, Object>) r.next().get("result");
+                    children = result.get("_children");
+                    childrenList = (List<Object>) children;
+                    childrenMap = ((Map<String, String>) childrenList.get(0));
+
+                    assertEquals("bk111", result.get("id"));
+                    assertEquals("author", childrenMap.get("_type"));
+                    assertEquals("O'Brien, Tim", childrenMap.get("_text"));
+                    assertEquals("title", ((Map) childrenList.get(1)).get("_type"));
+                    assertEquals("MSXML3: A Comprehensive Guide", ((Map) childrenList.get(1)).get("_text"));
+
+                    result = (Map<String, Object>) r.next().get("result");
+                    children = result.get("_children");
+                    childrenList = (List<Object>) children;
+                    childrenMap = ((Map<String, String>) childrenList.get(0));
+
+                    assertEquals("bk112", result.get("id"));
+                    assertEquals("author", childrenMap.get("_type"));
+                    assertEquals("Galos, Mike", childrenMap.get("_text"));
+                    assertEquals("title", ((Map) childrenList.get(1)).get("_type"));
+                    assertEquals("Visual Studio 7: A Comprehensive Guide", ((Map) childrenList.get(1)).get("_text"));
+                    assertEquals(false, r.hasNext());
+                });
+    }
+
+    @Test
+    public void testLoadXmlNoFailOnError() {
+        testCall(db, "CALL apoc.load.xml('books.xm', '', {failOnError:false}) yield value as result",
+                (r) -> {
+                    Map resultMap = (Map) r.get("result");
```

Comment:
Ma è giusto che faccia così? Non dovrebbe ritornare un emptyIterator?

Reply:
Si, lato codice:
`catch (FileNotFoundException e){
            if(!failOnError)
                return Stream.of(new MapResult(Collections.emptyMap()));
            else
                throw new FileNotFoundException(e.getMessage());
        }
        catch (Exception e){
            if(!failOnError)
                return Stream.of(new MapResult(Collections.emptyMap()));
            else
                throw new Exception(e);
        }
`
Nel caso nostro il file non esiste perche' e' stato passato errato